xbmc/guilib/DispResource.h

```diff
@@ -20,12 +20,21 @@
 
 #pragma once
 
+// interface for registering into windowing
+// to get notified about display events
+// interface only, does not control lifetime of the object
 class IDispResource
 {
 public:
-  virtual ~IDispResource() = default;
```

Comment:
Why have you deleted the dtor? If you want instances derived from this interface to get deleted correctly, even if you handle them using a reference to the interface (not the derived class), you need the virtual dtor.

IFace * i = new CDerivedFromIFace;
delete i; // will not call ~CDerivedFromIFace, ~IFace until you define virtual ~IFace()

Reply:
objects of this type are not supposed to be deleted. by having removed the dtor I stressed this. see my comment in the header

Reply:
I think it would be a good idea to make the destructor `protected` in this case so `delete i` in the example of @ksooo will not compile even if done by accident at some point.
See also https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rc-dtor-virtual

Reply:
Even if you remove  the dtor the compiler will still generate  a non-virtual one, if needed. So, this does not help to prevent object destruction.

Reply:
imo it is useless to prevent for something that can't happen anyway. nothing is completely foolproof and if you don't know how to use an interface and for what purpose, just don't use it. 
this is a callback interface used to get events. the holder of the interface does not control the lifetime of the object.

Reply:
Sorry, I don't get it. Aren't there any non-abstract classes derived from this interface? If so, there will be instances of these subclasses. And these instances will be created and necessarily destructed at some point in time, even if it is on application exit. So, how can it be that an instance is not supposed to be destructed?

Is the whole point that somebody getting a pointer to the interface must not / shall not destruct the instance behind the interface? If the answer is yes, then I would make the dtor protected.

Reply:
Yes, the holder of this interface is not supposed to either construct nor delete an object. Everbody taking a high level view on the design will see this. I won't make a brick jump off the wall if it is used incorrectly.

Reply:
It's all up to you.😄 I just wanted to understand your intention.